src/rust/engine/process_execution/src/docker.rs

```diff
@@ -735,11 +737,11 @@ impl ContainerCache {
 
   pub async fn shutdown(&self) -> Result<(), String> {
     // Skip shutting down if Docker was never initialized in the first place.
-    if !self.docker.initialized() {
+    if !DOCKER.initialized() {
       return Ok(());
     }
```

Comment:
If this is GLOBAL, using whether it has been initialized to determine whether to shut it down won't be accurate.

Instead, you might consider checking the length of `container_ids` _before_ `DOCKER.get`, and then bailing if it is empty?

Reply:
Good idea. Will do.